src/legacy/server/saved_objects/import/validate_references.ts

```diff
@@ -55,39 +61,66 @@ export async function validateReferences(
   savedObjects: SavedObject[],
   savedObjectsClient: SavedObjectsClient
 ) {
-  const errors: ImportError[] = [];
-
+  const errorMap: { [key: string]: ImportError } = {};
   const nonExistingReferenceKeys = await getNonExistingReferenceAsKeys(
     savedObjects,
     savedObjectsClient
   );
 
   // Filter out objects with missing references, add to error object
-  const filteredObjects = savedObjects.filter(savedObject => {
+  let filteredObjects = savedObjects.filter(savedObject => {
     const missingReferences = [];
-    for (const { type: refType, id: refId } of savedObject.references || []) {
-      if (!ENFORCED_TYPES.includes(refType)) {
-        continue;
-      }
+    const enforcedTypeReferences = (savedObject.references || []).filter(filterEnforcedTypes);
+    for (const { type: refType, id: refId } of enforcedTypeReferences) {
       if (nonExistingReferenceKeys.includes(`${refType}:${refId}`)) {
         missingReferences.push({ type: refType, id: refId });
       }
     }
-    if (missingReferences.length) {
-      errors.push({
-        id: savedObject.id,
-        type: savedObject.type,
-        error: {
-          type: 'missing_references',
-          references: missingReferences,
-        },
-      });
+    if (missingReferences.length === 0) {
+      return true;
+    }
+    errorMap[`${savedObject.type}:${savedObject.id}`] = {
+      id: savedObject.id,
+      type: savedObject.type,
+      title: savedObject.attributes && savedObject.attributes.title,
+      error: {
+        type: 'missing_references',
+        references: missingReferences,
+      },
+    };
+    return false;
+  });
+
+  // Filter out objects that reference objects within the import but are missing_references
+  // For example: visualization referencing a search that is missing an index pattern needs to be filtered out
+  filteredObjects = filteredObjects.filter(savedObject => {
+    let referenceErrors: MissingReferencesError['references'] = [];
+    const filteredReferences = (savedObject.references || []).filter(
+      ref =>
+        errorMap[`${ref.type}:${ref.id}`] &&
+        errorMap[`${ref.type}:${ref.id}`].error.type === 'missing_references'
+    );
+    for (const { type: refType, id: refId } of filteredReferences) {
+      const error = errorMap[`${refType}:${refId}`].error as MissingReferencesError;
+      referenceErrors = referenceErrors.concat(error.references);
+    }
+    if (referenceErrors.length === 0) {
+      return true;
     }
-    return missingReferences.length === 0;
+    errorMap[`${savedObject.type}:${savedObject.id}`] = {
+      id: savedObject.id,
+      type: savedObject.type,
+      title: savedObject.attributes && savedObject.attributes.title,
+      error: {
+        type: 'references_missing_references',
```

Comment:
If there's a better suggestion, let me know. This occurs when object references are failing to import due to `missing_references` error.